tests/expressions/tests.py

```diff
@@ -2333,10 +2334,46 @@ def test_resolve_output_field(self):
                 expr = Value(value)
                 self.assertIsInstance(expr.output_field, output_field_type)
 
-    def test_resolve_output_field_failure(self):
-        msg = "Cannot resolve expression type, unknown output_field"
-        with self.assertRaisesMessage(FieldError, msg):
-            Value(object()).output_field
+    def test_output_field_resolution(self):
+        # why does this fail, but test_resolve_output_field_with_null pass?
+        test_cases = [
+            Value(None),
+            Value("a string") + Value(None),
+            Value(42) + Value(None),
+            Value(datetime.date(2019, 5, 15)) + Value(None),
+            Value(datetime.time(3, 16)) + Value(None),
+            Value(datetime.datetime(2019, 5, 15)) + Value(None),
+            Value(datetime.timedelta(1)) + Value(None),
+            Value(Decimal("3.14")) + Value(None),
+            Value(b"") + Value(None),
+            Value(uuid.uuid4()) + Value(None),
+            Value(3.14159) + Value(None),  # doesn't throw error as expected
```

Comment:
to investigate why this doesn't throw an error and returns FloatField 

the SQL for it from experience should return null or pitch a fit: `SELECT (3.14152 + NULL) FROM TABLE`